pkg/provider/azure_zones.go

```diff
@@ -101,7 +101,7 @@ func (az *Cloud) getRegionZonesBackoff(ctx context.Context, region string) ([]*s
 	err := wait.ExponentialBackoffWithContext(ctx, az.RequestBackoff(), func(ctx context.Context) (done bool, err error) {
 		zones, innerErr = az.zoneRepo.ListZones(ctx)
 		if innerErr != nil {
-			klog.ErrorS(err, "Failed to list zones")
+			logger.Error(err, "Failed to list zones")
```

Comment:
not related to your change, but we should change it to innerErr here instead of err. please check why the unit test doesn't cover this bug if you have time.